app/uk/gov/hmrc/apiplatform/modules/approvals/services/ResponsibleIndividualVerificationService.scala

```diff
@@ -97,4 +131,18 @@ class ResponsibleIndividualVerificationService @Inject()(
       } yield riVerification
     ).value
   }
+
+  private def writeStateHistory(snapshotApp: ApplicationData, name: String) = 
+    insertStateHistory(snapshotApp, PENDING_GATEKEEPER_APPROVAL, Some(PENDING_RESPONSIBLE_INDIVIDUAL_VERIFICATION), name, COLLABORATOR, (a: ApplicationData) => applicationRepository.save(a))
+
+  private def insertStateHistory(snapshotApp: ApplicationData, newState: State, oldState: Option[State],
```

Comment:
This probably should not be duplicated in the 3 places (2 before you started).  Maybe.

Reply:
Moved method insertStateHistory into new base class